utils/build_profiler_report.py

```diff
@@ -0,0 +1,166 @@
+    if BaselineDict:
+        total_time_baseline = GetTotalTime(BaselineDict)
+
+    # sort targets by % taken of build time
+    temp = {k: v for k, v in sorted(CurrentDict.items(), key=lambda item: item[1], reverse=True)}
+    CurrentDict = temp
+
+    for target in CurrentDict.keys():
+        # percentage of build time that the target took
+        perc = CurrentDict[target]/total_time_current * 100
+
+        # difference between perc in current and in baseline
+        delta = 0
+        if BaselineDict:
+            if target_mismatch:
+                if target in BaselineDict.keys():
+                    delta = CurrentDict[target]/total_time_current_intersect * 100 - \
+                        BaselineDict[target]/total_time_baseline_intesect * 100
+
+            else:
+                delta = perc - (BaselineDict[target]/total_time_baseline * 100)
+
+        # time is the formatted build time of the target
+        time = FormatTime(CurrentDict[target])
+
+        # timeDelta a percentage difference of before and after build times 
+        if BaselineDict and target in BaselineDict.keys():
+            timeDelta = (CurrentDict[target]/BaselineDict[target]) * 100 - 100
+        else:
+            timeDelta = 0
+
+        line = [f'{target:60}', f"{perc:4.1f}", f"{delta:5.1f}", f"{time:8}", f"{timeDelta:+5.1f}"]
+        # if target was not in baseline, append note
+        if BaselineDict and target not in BaselineDict.keys():
+            line.append("Not in baseline")
+        print(' | '.join(line))
+
+    # Print time and % delta of the whole build time
+    if BaselineDict:
+        # totalDelta is the percentage change of build times between current and baseline
+        totalDelta = (total_time_current / total_time_baseline) * 100 - 100
+        line = ["-----\nTotal time:", FormatTime(total_time_current), "| %Δ", f'{totalDelta:+.1f}']
+        # if there are different targets in current and baseline log, add intersect delta, which
+        # compares build times while omitting conficting build targets
+        if target_mismatch:
+            intersectDelta = (total_time_current_intersect / total_time_baseline_intesect)\
+                 * 100 - 100
+            line.append(f'| intersect %Δ{intersectDelta:+5.1f}')
+        print(' '.join(line))
+    else:
+        print("-----\nTotal time:", FormatTime(total_time_current))
+
+    # Print targets which are present in baseline but not in current log
+    if BaselineDict:
+        removed = [target for target in BaselineDict.keys() if target not in CurrentDict.keys()]
+        print("-----\nTargets omitted from baseline:\n", '\n'.join(removed))
+
+
+def main() -> None:
+    # Dict key order used by PrintReport only specified in 3.7.0+
+    if sys.version_info < (3, 7, 0):
+        sys.stderr.write("You need python 3.7 or later to run this script\n")
+        exit(1)
+
+    if len(sys.argv) != 2:
+        print("Incorrect number of arguments")
+        sys.exit(99)
```

Comment:
Simply use argparse, it gives you standardized interface, meaningful separation of code into layers, and help in one package.